### DISCUSSION TITLE
[OKD4] [UPI vSphere / bare-metal] How to change in CoreOS the static network configuration's nameserver in a running cluster AFTER installation?

### DISCUSSION THREAD

**Discussion by devzeronull** [okd-project/okd#900] [Q&A]
Hi,
could anybody please tell me what the "desired" way is to change the CoreOS Node's DNS server in a running cluster or alter the static network configuration in general?

Referring to https://github.com/openshift/os/issues/400 which points to https://github.com/openshift/os/blob/master/docs/faq.md#q-how-do-i-provide-static-ip-addresses - overwriting "/etc/resolv.conf" through machineconfigs doesn't seem to be convenient, or?

If I understand https://github.com/openshift/os/blob/master/docs/faq.md#q-how-does-networking-differ-between-fedora-coreos-and-rhcos correctly, CoreOS doesn't make use of ifcfg NetworkManager scripts.

Looking into the FS on CoreOS it seems that configuration is made in scripts for the corresponding interfaces in:
```
/etc/NetworkManager/system-connections/ens192.nmconnection
/etc/NetworkManager/system-connections/br-ex.nmconnection
/etc/NetworkManager/system-connections/ovs-if-br-ex.nmconnection
/etc/NetworkManager/system-connections/ovs-port-br-ex.nmconnection
/etc/NetworkManager/system-connections/ovs-if-phys0.nmconnection
/etc/NetworkManager/system-connections/ovs-port-phys0.nmconnection
```
As suggested from the FAQ, would it also be okay to alter those scripts with a MC instead of writing to /etc/sysconfig/network-scripts?

We currently provide static network configuration through Boot Args on node provisioning which, if it would be the only way, would imply a cluster reinstallation since it is not possible to exchange master nodes...

Any hints or opinions would be highly appreciated!

Best regards
dzn

**Comment by kai-uwe-rommel** [comment]
If using vSphere UPI, you are supposed to be using the AfterBurn spec for static IP configuration, anyway.
Then it is easy to change that in the VM config at any time ... but it requires a VM reboot.
See https://docs.okd.io/latest/installing/installing_vsphere/installing-vsphere-network-customizations.html#installation-vsphere-machines_installing-vsphere-network-customizations and there step 5 g.

**Comment by forzamehlano** [comment]
Have you seen the nmstate operator? We use it for exactly the use case you describe:

https://docs.okd.io/latest/networking/k8s_nmstate/k8s-nmstate-updating-node-network-config.html#virt-example-nmstate-IP-management-dns_k8s_nmstate-updating-node-network-config
